src/Rings/MPolyMap/MPolyQuo.jl

```diff
@@ -0,0 +1,95 @@
+################################################################################
+#
+#  Domains of type MPolyQuo
+#
+################################################################################
+
+function _check_imgs_quo(R, S::NCRing, imgs)
+  for i in 1:n, j in 1:(i - 1)
+    @req imgs[i] * imgs[j] == imgs[j] * imgs[i] "Images $i and $j do not commute"
+  end
+  gensI = gens(modulus(R))
+  for g in gensI
+    @req iszero(evaluate(g, imgs)) "Morphism is not well-defined"
+  end
+  return nothing
+end
+
+# no check for commutative codomains
+function _check_imgs_quo(R, S::Ring, imgs) 
+  gensI = gens(modulus(R))
+  for g in gensI
+    @req iszero(evaluate(g, imgs)) "Morphism is not well-defined"
+  end
+end
+
+function hom(R::MPolyQuo, S::Ring, images::Vector; check::Bool = true)
+  n = ngens(R)
+  @req n == length(images) "Number of images must be $n"
+  # Now coerce into S or throw an error if not possible
+  imgs = _coerce(S, images)
+  if check
+    _check_imgs(S, imgs)
+    _check_homo(S, imgs)
+  end
+  return MPolyAnyMap(R, S, nothing, copy(imgs)) # copy because of #655
+end
+
+function hom(R::MPolyQuo, S::Ring, coeff_map, images::Vector; check::Bool = true)
+  n = ngens(R)
+  @req n == length(images) "Number of images must be $n"
+  # Now coerce into S or throw an error if not possible
+  imgs = _coerce(S, images)
+  if check
+    _check_imgs(S, imgs)
+    _check_homo(S, imgs)
+  end
+
+  return MPolyAnyMap(R, S, coeff_map, copy(imgs)) # copy because of #655
+end
+
+################################################################################
+#
+#  Evaluation
+#
+################################################################################
+
+function _evaluate_plain(F::MPolyAnyMap{<: MPolyQuo}, u)
+  return evaluate(lift(u), _images(F))
+end
+
+function _evaluate_general(F::MPolyAnyMap{<: MPolyQuo}, u)
+  if domain(F) === codomain(F)
+    return evaluate(map_coefficients(coefficient_map(F), lift(u),
+                                     parent = domain(F)), F.img_gens)
+  else
+    S = temp_ring(F)
+    if S !== nothing
+      evaluate(map_coefficients(coefficient_map(F), lift(u),
+                                parent = temp_ring), F.img_gens)
```

Comment:
So I have no idea about what this code does, so this may be utter nonsense (sorry!), but: do you perhaps means this?
```suggestion
                                parent = S), F.img_gens)
```